pkg/internal/ingress/convert_test.go

```diff
@@ -0,0 +1,157 @@
+					Number: 1234,
+				},
+			},
+		},
+		TLS: []networkingv1.IngressTLS{
+			{
+				Hosts:      []string{"aaa.", "bbb.", "ccc.ddd"},
+				SecretName: "test-secret-1",
+			},
+			{
+				Hosts:      []string{"eee"},
+				SecretName: "test-secret-2",
+			},
+		},
+		Rules: []networkingv1.IngressRule{
+			{
+				Host: "aaa",
+				IngressRuleValue: networkingv1.IngressRuleValue{
+					HTTP: &networkingv1.HTTPIngressRuleValue{
+						Paths: []networkingv1.HTTPIngressPath{
+							{
+								Path:     "/.well-known/acme-challenge",
+								PathType: func() *networkingv1.PathType { p := networkingv1.PathTypeImplementationSpecific; return &p }(),
+								Backend: networkingv1.IngressBackend{
+									Service: &networkingv1.IngressServiceBackend{
+										Name: "test-solver-backend",
+										Port: networkingv1.ServiceBackendPort{Number: 80},
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+	},
+}
+
+var v1beta1TestIngress = &networkingv1beta1.Ingress{
+	ObjectMeta: metav1.ObjectMeta{
+		Name:      "test-networkingv1-ingress",
+		Namespace: "test-networkingv1-namespace",
+		Annotations: map[string]string{
+			"test.key":                    "test.value",
+			"kubernetes.io/ingress.class": "bogus-ingress-class",
+		},
+		Labels: map[string]string{
+			"labelkey": "labelvalue",
+		},
+	},
+	Spec: networkingv1beta1.IngressSpec{
+		Backend: &networkingv1beta1.IngressBackend{
+			ServiceName: "default-backend-svc",
+			ServicePort: intstr.IntOrString{
+				Type:   intstr.Int,
+				IntVal: 1234,
+			},
+		},
+		Rules: []networkingv1beta1.IngressRule{
+			{
+				Host: "aaa",
+				IngressRuleValue: networkingv1beta1.IngressRuleValue{
+					HTTP: &networkingv1beta1.HTTPIngressRuleValue{
+						Paths: []networkingv1beta1.HTTPIngressPath{
+							{
+								Path:     "/.well-known/acme-challenge",
+								PathType: func() *networkingv1beta1.PathType { p := networkingv1beta1.PathTypeImplementationSpecific; return &p }(),
+								Backend: networkingv1beta1.IngressBackend{
+									ServiceName: "test-solver-backend",
+									ServicePort: intstr.IntOrString{
+										Type:   intstr.Int,
+										IntVal: 80,
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+		TLS: []networkingv1beta1.IngressTLS{
+			{
+				Hosts:      []string{"aaa.", "bbb.", "ccc.ddd"},
+				SecretName: "test-secret-1",
+			},
+			{
+				Hosts:      []string{"eee"},
+				SecretName: "test-secret-2",
+			},
+		},
+	},
+	Status: networkingv1beta1.IngressStatus{},
+}
+
+func TestConvert_networking_Ingress_To_v1beta1_Ingress(t *testing.T) {
+	in := v1TestIngress.DeepCopy()
+	out := &networkingv1beta1.Ingress{}
+	err := Convert_networking_Ingress_To_v1beta1_Ingress(in, out, nil)
+	assert.NoError(t, err, "converting networking v1 to networking v1beta1 Ingress should not fail")
+	expected := v1beta1TestIngress.DeepCopy()
+	assert.Equal(t, expected, out, "Conversion from networking v1 to networking v1beta1 Ingress was not as expected")
+}
```

Comment:
nitpick: am I correct in saying that this test misses the case where `out` doesn't have any annotations and therefore `Convert_networking_Ingress_To_v1beta1_Ingress` has an additional branch where it runs `out.Annotations = make(map[string]string)`? if so, would it be worth adding a test which covers that branch?